src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1359,6 +1359,12 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotificatio
         break;
     }
 
+    if (ThreadStackMgrImpl().mIsSrpClearAllRequested)
```

Comment:
How about executing that under `aHostInfo->mState == OT_SRP_CLIENT_ITEM_STATE_REMOVED` condition above in case another SRP update was completed just before the factory reset?

Reply:
True I placed it there before and moved it down when testing something. I'll move it back up thanks !